### PR TITLE
refactor(python): Expose transform as a submodule for pyiceberg_core

### PR DESCRIPTION
**Pull request by Xuanwo**
This PR will expose transform as a submodule for pyiceberg_core.

Also, this PR removes not useful hello_world function.

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi, @sungwy, would you like to take a review? I believe this way is more pythonic.

**Comment by Xuanwo** [comment]
> There's an outstanding issue with pyo3 that prevents the functions defined within a module to be imported like below:

Oh, I see the problem. Let me take a look.

**Comment by Xuanwo** [comment]
Hi, @sungwy, I fixed this issue and add a test for it. Please take a look, thanks!

**Review by sungwy — APPROVED** [review]
Oh man that's amazing. Thank you for the fix!

File: bindings/python/src/transform.rs [hunk 1]
```diff
@@ -15,24 +15,55 @@
...
+#[pyfunction]
+pub fn bucket(py: Python, array: PyObject, num_buckets: u32) -> PyResult<PyObject> {
+    apply(py, array, Transform::Bucket(num_buckets))
+}
+
+#[pyfunction]
+pub fn truncate(py: Python, array: PyObject, width: u32) -> PyResult<PyObject> {
+    apply(py, array, Transform::Truncate(width))
+}
+
```
Comment (kevinjqliu) [inline bindings/python/src/transform.rs:65]: should we also include the `unknown` transform? 
https://github.com/apache/iceberg/blob/8d97d54756a1b8ba1986858b2461013864b9e37f/core/src/main/java/org/apache/iceberg/Partitioning.java#L50-L103

**Review by kevinjqliu — APPROVED** [review]
this looks great, thanks for the refactor! added a minor comment. 
I'll also refactor #604 to follow this format.

File: bindings/python/src/transform.rs (hunk 1 above)
Comment (Xuanwo) [reply bindings/python/src/transform.rs:65]: Maybe we can add them in follow up PRs?

Comment (kevinjqliu) [reply bindings/python/src/transform.rs:65]: sg!

**Comment by Xuanwo** [comment]
Thank you @sungwy and @kevinjqliu for the quick review. We can implement more transforms in following PRs. And I will merge this one first 😘

**Pull request merged**